lambdas/data-migration2/main.tf

```diff
@@ -0,0 +1,106 @@
+locals {
+  lambda_path = "${path.module}/dist/webpack/lambda.zip"
+}
+
+data "aws_iam_policy_document" "lambda_assume_role_policy" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "data_migration2" {
+  name                 = "${var.prefix}-data-migration2"
+  assume_role_policy   = data.aws_iam_policy_document.lambda_assume_role_policy.json
+  permissions_boundary = var.permissions_boundary_arn
+
+  tags = var.tags
+}
+
+data "aws_iam_policy_document" "data_migration2" {
+  statement {
+    actions = [
+      "ec2:CreateNetworkInterface",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:DescribeLogStreams",
+      "logs:PutLogEvents"
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "dynamodb:Scan",
+    ]
+    resources = [
+      var.dynamo_tables.executions.arn
+    ]
+  }
+
+  statement {
+    actions = [
+      "secretsmanager:GetSecretValue"
+    ]
+    resources = [var.rds_user_access_secret_arn]
+  }
+}
+
+resource "aws_iam_role_policy" "data_migration2" {
+  name   = "${var.prefix}_data_migration2"
+  role   = aws_iam_role.data_migration2.id
+  policy = data.aws_iam_policy_document.data_migration2.json
+}
+
+resource "aws_security_group" "data_migration2" {
+  count = length(var.lambda_subnet_ids) == 0 ? 0 : 1
+
+  name   = "${var.prefix}-data-migration"
+  vpc_id = var.vpc_id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = var.tags
+}
+
+resource "aws_lambda_function" "data_migration2" {
+  function_name    = "${var.prefix}-data-migration2"
+  filename         = local.lambda_path
+  source_code_hash = filebase64sha256(local.lambda_path)
+  handler          = "index.handler"
+  role             = aws_iam_role.data_migration2.arn
+  runtime          = "nodejs12.x"
+  timeout          = 300
+  memory_size      = 512
```

Comment:
We should consider doing an assessment on mem/performance costs vs disk reads.    If it's linear and the DB can handle it we should consider upping this or possibly making it tuneable.

Reply:
Also if we don't want this lambda to run independently, it might be worth explicitly disabling this or lowering the timeout.

Reply:
To be addressed in another ticket. Jira # forthcoming.